src/components/HeaderTodoApp/HeaderTodoApp.tsx

```diff
@@ -0,0 +1,52 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, {
+  FC, FormEvent, useState,
+} from 'react';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todos: Todo[];
+  setError: (error: string) => void;
+}
+
+export const HeaderTodoApp: FC<Props> = React.memo(({
+  todos,
+  setError,
+}) => {
+  const [query, setQuery] = useState('');
```

Comment:
`query` not the best name in this case and used more for search queries
if you need hold some name call it close to `name` or `title`